transport-classes-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollChannel.java

```diff
@@ -521,39 +507,14 @@ public void handle(IoRegistration registration, IoEvent event) {
          */
         abstract void epollInReady();
 
-        final void epollInBefore() {
-            maybeMoreDataToRead = false;
-        }
-
-        final void epollInFinally(ChannelConfig config) {
-            maybeMoreDataToRead = allocHandle.maybeMoreDataToRead();
-
-            if (allocHandle.isReceivedRdHup() || (readPending && maybeMoreDataToRead)) {
-                // trigger a read again as there may be something left to read and because of epoll ET we
-                // will not get notified again until we read everything from the socket
-                //
-                // It is possible the last fireChannelRead call could cause the user to call read() again, or if
-                // autoRead is true the call to channelReadComplete would also call read, but maybeMoreDataToRead is set
-                // to false before every read operation to prevent re-entry into epollInReady() we will not read from
-                // the underlying OS again unless the user happens to call read again.
-                executeEpollInReadyRunnable(config);
-            } else if (!readPending && !config.isAutoRead()) {
-                // Check if there is a readPending which was not processed yet.
-                // This could be for two reasons:
-                // * The user called Channel.read() or ChannelHandlerContext.read() in channelRead(...) method
-                // * The user called Channel.read() or ChannelHandlerContext.read() in channelReadComplete(...) method
-                //
-                // See https://github.com/netty/netty/issues/2254
-                clearEpollIn();
-            }
-        }
-
-        final void executeEpollInReadyRunnable(ChannelConfig config) {
-            if (epollInReadyRunnablePending || !isActive() || shouldBreakEpollInReady(config)) {
-                return;
-            }
-            epollInReadyRunnablePending = true;
-            eventLoop().execute(epollInReadyRunnable);
+        boolean shouldStopReading(ChannelConfig config) {
```

Comment:
nit: final?

Reply:
it was complaining when I add final as people might already extend it and have such a definition. I guess its something that is very unlikely so we could add it if we want

Reply:
wat... I say make it final. If someone has already defined such a method (seems extremely unlikely) it may not behave the same so we'd want to know.

Reply:
done